WIP : egs/aspire : added nnet3 xent, sMBR and chain recipes

vijayaditya opened the pull request:
Has generic scripts to copy and combine lattice directories after adding a prefix or suffix. 
Some of the recipe scripts have not been updated.

I will add results once the experiments are complete.

vijayaditya commented on egs/aspire/s5/local/chain/copy_lat_dir.sh:
```diff
@@ -0,0 +1,66 @@
+
+$cmd JOB=1:$nj $dest_dir/log/copy_lat.JOB.log \
+  lattice-copy "ark:gunzip -c $src_dir/lat.JOB.gz |" ark,t:- \|\
+    awk -v prefix="$utt_prefix" -v suffix="$utt_suffix" '{if ((NF == 1) && (prev_nf == 0)) { printf("%s%s%s\n", prefix, $0, suffix) } else { print $0}; prev_nf=NF}' \|\
```
@danpovey , @jtrmal To identify the utterance name lines in the lattices I am checking that the number of fields in the current line is 1 and the number of fields in the previous line is 0. This works for lattices currently copied by `lattice-copy`. Is this a safe assumption to make ? If not, is there any other way to affix strings to utt-ids other than adding these options to  `lattice-copy` ?

vijayaditya replied:
@danpovey When copying a lattice directory which files should be definitely be copied other than `lat.*.gz tree cmvn_opts splice_opts num_jobs final.mdl` ?

vijayaditya replied:
@tomkocse You could use these scripts when writing the multi-condition multi-en recipe ( #870 ).

danpovey replied:
Yes, this assumption about identifying the utterance-ids is safe.  An alternative method is to dump lattices to ark,scp (without gzipping them) and to do operations on the scp file instead.
If you copy the splice_opts you should probably copy delta_opts too (although neither of them is guaranteed to exist).

vijayaditya commented:
```
nnet2: 
%WER 30.8 | 2120 27213 | 75.3 16.2 8.4 6.2 30.8 78.8 | -0.724 | exp/nnet2_multicondition/nnet_ms_a/decode_dev_aspire_whole_uniformsegmented_win10_over5_v6_iterfinal_pp_fg/score_13/penalty_0.0/ctm.filt.filt.sys

nnet3:
%WER 31.0 | 2120 27217 | 74.8 16.1 9.1 5.9 31.0 77.9 | -0.707 | exp/nnet3/tdnn/decode_dev_aspire_whole_uniformsegmented_win10_over5_v6_200jobs_iterfinal_pp_fg/score_14/penalty_0.0/ctm.filt.filt.sys
```

vijayaditya commented:
The BLSTMs show significantly better frame accuracy (see fig below) but the WERs are worse.

```
iteration	TDNN	        BLSTM
====================		
100	        34.2	          43.8
200	        32.8	         40.4
300	        32.3	         38.3
400	        31.7	         36.8
500	        31.6	         35.8
600	        31.3	
700	        31.2	
800	        31.1	
900	          ----
1000	31.1	
Final         31	
``` 

![screen shot 2016-07-14 at 11 14 38 pm](https://cloud.githubusercontent.com/assets/3895456/16862622/0a6f95fa-4a19-11e6-8b0d-77d280defa3f.png)

naxingyu commented:
What's the scale of aspire? BLSTM might need more epochs in large scale 
recipes, as in fisher_swbd.


在 2016/7/15 11:17, Vijayaditya Peddinti 写道:
>
> The BLSTMs show significantly better frame accuracy (see fig below) 
> but the WERs are worse.
>
> |iteration TDNN BLSTM ==================== 100 34.2 43.8 200 32.8 40.4 
> 300 32.3 38.3 400 31.7 36.8 500 31.6 35.8 600 31.3 700 31.2 800 31.1 
> 900 ---- 1000 31.1 Final 31 |
>
> screen shot 2016-07-14 at 11 14 38 pm 
> <https://cloud.githubusercontent.com/assets/3895456/16862622/0a6f95fa-4a19-11e6-8b0d-77d280defa3f.png>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/875#issuecomment-232850451>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ADKpxJLJ7piCqFrr2cCCD9NXcN4WqeWeks5qVvvbgaJpZM4JCR8D>.
>

vijayaditya commented:
These recipes use the same hyper-parameters as fshr_swbd. Only 1/4th of the iterations are complete in BLSTM. However my concern is that even though BLSTM has better train/valid accuracies its WER performance is very poor.

vijayaditya commented:
The BLSTM training logs several red flags. The average parameter change is around 50 (`--max-param-change 2`) for the first 800/2000 iterations. I also see a huge dip in the objfn around iteration 920. However going through the logs I see no major event (e.g. sudden increase in parameter-change). 

Log dir: /export/b13/vpeddinti/lsr/egs/aspire/s5_olddataaugment/exp/nnet3/lstm_bidirectional/log/

![screen shot 2016-07-18 at 1 58 56 am](https://cloud.githubusercontent.com/assets/3895456/16906805/42cbe178-4c8b-11e6-81bc-7da418188edc.png)
![screen shot 2016-07-18 at 1 59 07 am](https://cloud.githubusercontent.com/assets/3895456/16906808/45940386-4c8b-11e6-8d5f-37adb2004e2f.png)

vijayaditya commented:
Nice improvement with chain models in ASpIRE.

```bash 
chain (900/1600 iterations): 
%WER 28.2 | 2120 27211 | 77.4 13.4 9.2 5.6 28.2 75.7 | -0.611 | exp/chain/tdnn_7b/decode_dev_aspire_whole_uniformsegmented_win10_over5_v6_200jobs_frshifted_iter900_pp_fg/score_9/penalty_0.25/ctm.filt.filt.sys

xent (final model ~1100 iterations):
%WER 31.0 | 2120 27217 | 74.8 16.1 9.1 5.9 31.0 77.9 | -0.707 | exp/nnet3/tdnn/decode_dev_aspire_whole_uniformsegmented_win10_over5_v6_200jobs_iterfinal_pp_fg/score_14/penalty_0.0/ctm.filt.filt.sys

```

vijayaditya commented:
@danpovey Ready for review.

```bash 
chain (Final model ~1600 iterations): 
%WER 27.8 | 2120 27217 | 78.2 13.6 8.2 6.0 27.8 75.9 | -0.613 | exp/chain/tdnn_7b/decode_dev_aspire_whole_uniformsegmented_win10_over5_v6_200jobs_iterfinal_pp_fg/score_9/penalty_0.0/ctm.filt.filt.sys

xent (final model ~1100 iterations):
%WER 31.0 | 2120 27217 | 74.8 16.1 9.1 5.9 31.0 77.9 | -0.707 | exp/nnet3/tdnn/decode_dev_aspire_whole_uniformsegmented_win10_over5_v6_200jobs_iterfinal_pp_fg/score_14/penalty_0.0/ctm.filt.filt.sys

```


Note: I have added the BLSTM script but there are not results available as the experiments were paused for Babel evals.

danpovey commented on egs/aspire/s5/run.sh:
```diff
@@ -156,6 +158,7 @@ steps/train_sat.sh  --cmd "$train_cmd" \
 
 # build silprob lang directory
 local/build_silprob.sh
+exit;
```
was this supposed to be temporary?

vijayaditya replied:
Yes, thanks for pointing out.

danpovey commented on egs/wsj/s5/utils/combine_lat_dirs.sh:
```diff
@@ -0,0 +1,100 @@
+#                 Vijay Peddinti
+
+# This script operates on lattice directories, such as exp/tri4a_lats
+
```
please add a line or two saying what it does and what its output[s] are.

danpovey replied:
Also, the general rule is that any script that takes the 'cmd' option should be in steps/.

danpovey commented on egs/wsj/s5/utils/combine_lat_dirs.sh:
```diff
@@ -0,0 +1,100 @@
+echo $num_jobs > $dest/num_jobs  || exit 1
+
+echo "$0: splitting the lattices to appropriate chunks according to the reference utt2spk files."
+for i in `seq 1 $num_jobs`; do
```
Here,  you should be able to use filter_scps.pl, which is more efficient.

danpovey replied:
also, there is no need to create the utt_subset files.  filter_scp.pl and filter_scps.pl automatically take the first field of the filter file.

vijayaditya replied:
Sorry I based this script on `utils/combine_ali_dirs.sh` just replacing alignment commands with lattice commands. I will add the suggested changes to both these scripts.

danpovey replied:
@xiaohui-zhang, please remember these comments on the script
combine_ali_dirs.sh for the future, although Vijay will fix it.

Where do you use that script?  I can't find anything that uses it.


On Thu, Jul 21, 2016 at 2:35 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/wsj/s5/utils/combine_lat_dirs.sh
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71791586>:
>
> > +for dir in $*; do
> > +  src_id=$((src_id + 1))
> > +  cur_num_jobs=$(cat $dir/num_jobs) || exit 1;
> > +  all_ids=$(seq -s, $cur_num_jobs)
> > +  $cmd $dir/log/copy_lattices.log \
> > +    lattice-copy "ark:gunzip -c $dir/lat.{$all_ids}.gz|" \
> > +    ark,scp:$temp_dir/lat.$src_id.ark,$temp_dir/lat.$src_id.scp || exit 1;
> > +done
> > +cat $temp_dir/lat.*.scp | sort -m > $temp_dir/lat.scp || exit 1;
> > +
> > +echo "$0: splitting data to get reference utt2spk for individual lat.JOB.gz files."
> > +utils/split_data.sh $data $num_jobs || exit 1;
> > +echo $num_jobs > $dest/num_jobs  || exit 1
> > +
> > +echo "$0: splitting the lattices to appropriate chunks according to the reference utt2spk files."
> > +for i in `seq 1 $num_jobs`; do
>
> Sorry I based this script on utils/combine_ali_dirs.sh just replacing
> alignment commands with lattice commands. I will add the suggested changes
> to both these scripts.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/875/files/66ed5e218d617413a0a197737dea5c494d346547#r71791586>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVuzY3Gym3U18RyB9elBe3zWF-aYBtks5qX-YTgaJpZM4JCR8D>
> .
>

xiaohui-zhang replied:
Thanks. Hmm it's being used in babel multi-condition experiments . (will commit it in several days)

danpovey replied:
Vijay, I notice that in egs/aspire/s5/local/multi_condition/ you have your
own combine_ali_dirs.sh script, and it's very hard to understand with a big
inline python section.  I think it would be better to have a single
definitive script to do this, located in steps/.

On Thu, Jul 21, 2016 at 2:45 PM, Daniel Povey <dpovey@gmail.com> wrote:

> @xiaohui-zhang, please remember these comments on the script
> combine_ali_dirs.sh for the future, although Vijay will fix it.
>
> Where do you use that script?  I can't find anything that uses it.
>
>
> On Thu, Jul 21, 2016 at 2:35 PM, Vijayaditya Peddinti <
> notifications@github.com> wrote:
>
>> In egs/wsj/s5/utils/combine_lat_dirs.sh
>> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71791586>:
>>
>> > +for dir in $*; do
>> > +  src_id=$((src_id + 1))
>> > +  cur_num_jobs=$(cat $dir/num_jobs) || exit 1;
>> > +  all_ids=$(seq -s, $cur_num_jobs)
>> > +  $cmd $dir/log/copy_lattices.log \
>> > +    lattice-copy "ark:gunzip -c $dir/lat.{$all_ids}.gz|" \
>> > +    ark,scp:$temp_dir/lat.$src_id.ark,$temp_dir/lat.$src_id.scp || exit 1;
>> > +done
>> > +cat $temp_dir/lat.*.scp | sort -m > $temp_dir/lat.scp || exit 1;
>> > +
>> > +echo "$0: splitting data to get reference utt2spk for individual lat.JOB.gz files."
>> > +utils/split_data.sh $data $num_jobs || exit 1;
>> > +echo $num_jobs > $dest/num_jobs  || exit 1
>> > +
>> > +echo "$0: splitting the lattices to appropriate chunks according to the reference utt2spk files."
>> > +for i in `seq 1 $num_jobs`; do
>>
>> Sorry I based this script on utils/combine_ali_dirs.sh just replacing
>> alignment commands with lattice commands. I will add the suggested changes
>> to both these scripts.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kaldi-asr/kaldi/pull/875/files/66ed5e218d617413a0a197737dea5c494d346547#r71791586>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ADJVuzY3Gym3U18RyB9elBe3zWF-aYBtks5qX-YTgaJpZM4JCR8D>
>> .
>>
>
>

danpovey commented on egs/wsj/s5/utils/combine_lat_dirs.sh:
```diff
@@ -0,0 +1,100 @@
+    lattice-copy "ark:gunzip -c $dir/lat.{$all_ids}.gz|" \
+    ark,scp:$temp_dir/lat.$src_id.ark,$temp_dir/lat.$src_id.scp || exit 1;
+done
+cat $temp_dir/lat.*.scp | sort -m > $temp_dir/lat.scp || exit 1;
```
sort -m doesn't work like this, you have to give it the individual inputs, like
sort -m $temp_dir/lat.*.scp

vijayaditya commented:
I am waiting for #706 to be merged. After this I will be updating the data preparation code and alignment generation code.

vijayaditya commented:
Deleted `local/multi_condition/combine_ali_dirs.sh`, modified `steps/combine_ali_dirs.sh` and deleted `steps/combine_lat_dirs.sh` as I turned to be unnecessary for the recipe.

danpovey commented on egs/wsj/s5/utils/combine_ali_dirs.sh:
```diff
@@ -1,99 +0,0 @@
-if [[ $# -lt 3 ]]; then
-  echo "Usage: $0 [options] <data> <dest-ali-dir> <src-ali-dir1> <src-ali-dir2> ..."
-  echo "e.g.: $0 --num-jobs 32 data/train exp/tri3_ali_combined exp/tri3_ali_1 exp_tri3_ali_2"
-  echo "Options:"
```
I think you forgot to 'git add' this in its location in steps.

vijayaditya replied:
Sorry about that, the file is available now.

On Thu, Jul 21, 2016 at 7:39 PM, Daniel Povey <notifications@github.com>
wrote:

> In egs/wsj/s5/utils/combine_ali_dirs.sh
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71806911>:
>
> > -# This srcipt operates on alignment directories, such as exp/tri4a_ali
> > -
> > -# Begin configuration section.
> > -cmd=run.pl
> > -extra_files=
> > -num_jobs=4
> > -# End configuration section.
> > -echo "$0 $@"  # Print the command line for logging
> > -
> > -if [ -f path.sh ]; then . ./path.sh; fi
> > -. parse_options.sh || exit 1;
> > -
> > -if [[ $# -lt 3 ]]; then
> > -  echo "Usage: $0 [options] <data> <dest-ali-dir> <src-ali-dir1> <src-ali-dir2> ..."
> > -  echo "e.g.: $0 --num-jobs 32 data/train exp/tri3_ali_combined exp/tri3_ali_1 exp_tri3_ali_2"
> > -  echo "Options:"
>
> I think you forgot to 'git add' this in its location in steps.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/875/files/c9298ce792414a31f45da61a734afd38c77426f5#r71806911>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADtwoNZe9fTJ6BFvMHxqX06t2JqKZT6Vks5qYANMgaJpZM4JCR8D>
> .
>

danpovey commented on egs/wsj/s5/steps/combine_ali_dirs.sh:
```diff
@@ -61,6 +62,7 @@ done
 
 src_id=0
 temp_dir=$dest/temp
+rm -rf $temp_dir
```
No -f flag on rm, please, better to do
[ -d $temp_dir ] && rm -r $temp_dir
I just don't like using the -f flag on 'rm' as a general policy.

danpovey commented on egs/wsj/s5/steps/combine_ali_dirs.sh:
```diff
@@ -71,23 +73,25 @@ for dir in $*; do
 
 echo "$0: splitting data to get reference utt2spk for individual ali.JOB.gz files."
 utils/split_data.sh $data $num_jobs || exit 1;
-echo $num_jobs > $dest/num_jobs  || exit 1
```
There may be situations where we need this 'num_jobs'... better to keep that.

danpovey commented on egs/wsj/s5/steps/combine_ali_dirs.sh:
```diff
@@ -71,23 +73,25 @@ for dir in $*; do
+    copy-int-vector scp:$temp_dir/ali.${i}.scp "ark:|gzip -c >$dest/ali.$i.gz" || exit 1;
+done
+
+echo $num_jobs > $dest/num_jobs  || exit 1
```
.. it's OK, I see you are doing it here.

danpovey commented on egs/wsj/s5/steps/nnet3/decode.sh:
```diff
@@ -80,7 +80,7 @@ thread_string=
 
 mkdir -p $dir/log
-[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj || exit 1;
+[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh --per-utt $data $nj || exit 1;
```
Was the addition of this --per-utt flag intended to be temporary?
Until issue #778 is addressed, this is not safe to commit as it would interfere with jobs that do require splitting per-speaker.

vijayaditya replied:
I intended it to be permanent as the ASpIRE decodes take several hours when
done per speaker. I will remove it from the script as soon as I get access
to a computer.

Vijay

On Jul 21, 2016 8:08 PM, "Daniel Povey" <notifications@github.com> wrote:

> In egs/wsj/s5/steps/nnet3/decode.sh
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71809522>:
>
> > @@ -80,7 +80,7 @@ thread_string=
> >  [ $num_threads -gt 1 ] && thread_string="-parallel --num-threads=$num_threads"
> >
> >  mkdir -p $dir/log
> > -[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj || exit 1;
> > +[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh --per-utt $data $nj || exit 1;
>
> Was the addition of this --per-utt flag intended to be temporary?
> Until issue #778 <https://github.com/kaldi-asr/kaldi/issues/778> is
> addressed, this is not safe to commit as it would interfere with jobs that
> do require splitting per-speaker.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/875/files/2a81f157021f8b7adf2d175b945cc40b4e794e6e#r71809522>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADtwoLg2PCygEZLJlaVHee-SM5h61RNmks5qYAntgaJpZM4JCR8D>
> .
>

danpovey replied:
We've been waiting too long for the per-utt splitting to be working.
I'll change the data-splitting script right now and check it in within the
hour.
After that you can set something like
sdata=$dir/split${nj}utt
instead of
sdata=$dir/split${nj}
and it will work.
Dan



On Thu, Jul 21, 2016 at 5:19 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/wsj/s5/steps/nnet3/decode.sh
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71810489>:
>
> > @@ -80,7 +80,7 @@ thread_string=
> >  [ $num_threads -gt 1 ] && thread_string="-parallel --num-threads=$num_threads"
> >
> >  mkdir -p $dir/log
> > -[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj || exit 1;
> > +[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh --per-utt $data $nj || exit 1;
>
> I intended it to be permanent as the ASpIRE decodes take several hours
> when done per speaker. I will remove it from the script as soon as I get
> access to a computer. Vijay
> … <#m_-858257127955897913_>
> On Jul 21, 2016 8:08 PM, "Daniel Povey" ***@***.***> wrote: In
> egs/wsj/s5/steps/nnet3/decode.sh <#875 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71809522>>: > @@
> -80,7 +80,7 @@ thread_string= > [ $num_threads -gt 1 ] &&
> thread_string="-parallel --num-threads=$num_threads" > > mkdir -p $dir/log
> > -[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj
> || exit 1; > +[[ -d $sdata && $data/feats.scp -ot $sdata ]] ||
> split_data.sh --per-utt $data $nj || exit 1; Was the addition of this
> --per-utt flag intended to be temporary? Until issue #778
> <https://github.com/kaldi-asr/kaldi/issues/778> <#778
> <https://github.com/kaldi-asr/kaldi/issues/778>> is addressed, this is
> not safe to commit as it would interfere with jobs that do require
> splitting per-speaker. — You are receiving this because you authored the
> thread. Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/875/files/2a81f157021f8b7adf2d175b945cc40b4e794e6e#r71809522>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/ADtwoLg2PCygEZLJlaVHee-SM5h61RNmks5qYAntgaJpZM4JCR8D>
> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/875/files/2a81f157021f8b7adf2d175b945cc40b4e794e6e#r71810489>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu_eKrztA8tWJ7xCcwHOFlmxoA8G7ks5qYAyYgaJpZM4JCR8D>
> .
>

vijayaditya replied:
OK thanks.

Vijay

On Jul 21, 2016 8:21 PM, "Daniel Povey" <notifications@github.com> wrote:

> In egs/wsj/s5/steps/nnet3/decode.sh
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71810668>:
>
> > @@ -80,7 +80,7 @@ thread_string=
> >  [ $num_threads -gt 1 ] && thread_string="-parallel --num-threads=$num_threads"
> >
> >  mkdir -p $dir/log
> > -[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj || exit 1;
> > +[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh --per-utt $data $nj || exit 1;
>
> We've been waiting too long for the per-utt splitting to be working. I'll
> change the data-splitting script right now and check it in within the hour.
> After that you can set something like sdata=$dir/split${nj}utt instead of
> sdata=$dir/split${nj} and it will work. Dan On Thu, Jul 21, 2016 at 5:19
> PM, Vijayaditya Peddinti < notifications@github.com> wrote:
> … <#m_-6281962373127726343_>
> In egs/wsj/s5/steps/nnet3/decode.sh <#875 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71810489>>: > @@
> -80,7 +80,7 @@ thread_string= > [ $num_threads -gt 1 ] &&
> thread_string="-parallel --num-threads=$num_threads" > > mkdir -p $dir/log
> > -[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj
> || exit 1; > +[[ -d $sdata && $data/feats.scp -ot $sdata ]] ||
> split_data.sh --per-utt $data $nj || exit 1; I intended it to be permanent
> as the ASpIRE decodes take several hours when done per speaker. I will
> remove it from the script as soon as I get access to a computer. Vijay …
> <#m_-858257127955897913_> On Jul 21, 2016 8:08 PM, "Daniel Povey"
> ***@***.***> wrote: In egs/wsj/s5/steps/nnet3/decode.sh <#875
> <https://github.com/kaldi-asr/kaldi/pull/875> (comment) <#875 (comment)
> <https://github.com/kaldi-asr/kaldi/pull/875#discussion_r71809522>>>: >
> @@ -80,7 +80,7 @@ thread_string= > [ $num_threads -gt 1 ] &&
> thread_string="-parallel --num-threads=$num_threads" > > mkdir -p $dir/log
> > -[[ -d $sdata && $data/feats.scp -ot $sdata ]] || split_data.sh $data $nj
> || exit 1; > +[[ -d $sdata && $data/feats.scp -ot $sdata ]] ||
> split_data.sh --per-utt $data $nj || exit 1; Was the addition of this
> --per-utt flag intended to be temporary? Until issue #778
> <https://github.com/kaldi-asr/kaldi/issues/778> <#778
> <https://github.com/kaldi-asr/kaldi/issues/778>> <#778
> <https://github.com/kaldi-asr/kaldi/issues/778> <#778
> <https://github.com/kaldi-asr/kaldi/issues/778>>> is addressed, this is
> not safe to commit as it would interfere with jobs that do require
> splitting per-speaker. — You are receiving this because you authored the
> thread. Reply to this email directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/875/files/2a81f157021f8b7adf2d175b945cc40b4e794e6e#r71809522>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/ADtwoLg2PCygEZLJlaVHee-SM5h61RNmks5qYAntgaJpZM4JCR8D>
> . — You are receiving this because you were mentioned. Reply to this email
> directly, view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/875/files/2a81f157021f8b7adf2d175b945cc40b4e794e6e#r71810489>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/ADJVu_eKrztA8tWJ7xCcwHOFlmxoA8G7ks5qYAyYgaJpZM4JCR8D>
> .
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/875/files/2a81f157021f8b7adf2d175b945cc40b4e794e6e#r71810668>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADtwoN_YYDfMEGv6B0fPPrZcE4jwBCsEks5qYA0kgaJpZM4JCR8D>
> .
>

danpovey replied:
ok, committed that change.

danpovey replied:
You now have to change `sdata=$data/split$nj;` to `sdata=$data/split${nj}utt;`

danpovey commented on egs/wsj/s5/steps/nnet3/decode.sh:
```diff
@@ -80,7 +80,7 @@ thread_string=
 [ $num_threads -gt 1 ] && thread_string="-parallel --num-threads=$num_threads"
```
vijay, I think we can merge this once you make this small change... a few lines above here, add 'utt' at the end of 'sdata'.

vijayaditya commented:
@danpovey All the requested changes have been done. Sorry for the delay, I was unable to test out the final changes due to IO latency issues on CLSP cluster till now.

The recipe in the current form is not going to be able to reproduce the results due to the bug in `wav-reverberate` command. This bug has been corrected in #706 , which also does other major modifications to the data reverberation scripts. So I am not going to address this issue in this PR.
